addons/sale_product_configurator/static/src/js/product_configurator_widget.js

```diff
@@ -99,40 +77,36 @@ var ProductConfiguratorWidget = relationalFields.FieldMany2One.extend({
                         },
                         onSuccess: function () {
                             if (result.has_optional_products) {
-                                self._openProductConfigurator({
-                                        configuratorMode: 'options',
-                                        default_pricelist_id: self._getPricelistId(),
-                                        default_product_template_id: productTemplateId
-                                    },
-                                    event.data.dataPointID
-                                );
+                                self._openOptionsConfigurator(productTemplateId, ev.data.dataPointID);
                             }
+                            // TODO should be called before the _openOptionsConfigurator in the future
```

Comment:
_onProductChange is called by the field_changed event triggered.

The problem here is that the options configuration should ideally happen after the _onProductChange because currently, if you have an event with optional products, the two wizards are opened at the same time, which leads to unexpected results.